pyteal/ast/base64decode.py

```diff
@@ -0,0 +1,85 @@
+from typing import TYPE_CHECKING
+from enum import Enum
+
+from pyteal.types import TealType, require_type
+from pyteal.errors import verifyFieldVersion
+from pyteal.ir import TealOp, Op, TealBlock
+from pyteal.ast.expr import Expr
+from pyteal.ast.leafexpr import LeafExpr
+
+if TYPE_CHECKING:
+    from pyteal.compiler import CompileOptions
+
+
+class Base64Encoding(Enum):
+    # fmt: off
+    #     id  |   name   | min version
+    url = (0, "URLEncoding", 7)
+    std = (1, "StdEncoding", 7)
+    # fmt: on
+
+    def __init__(self, id: int, name: str, min_version: int) -> None:
+        self.id = id
+        self.arg_name = name
+        self.min_version = min_version
+
+
+Base64Encoding.__module__ = "pyteal"
+
+
+class Base64Decode(LeafExpr):
+    """An expression that decodes a base64-encoded byte string according to a specific encoding.
+
+    See [RFC 4648](https://rfc-editor.org/rfc/rfc4648.html#section-4) (sections 4 and 5) for information on specifications.
+
+    It is assumed that the encoding ends with the exact number of = padding characters as required by the RFC.
```

Comment:
Perhaps add a warning to NOT use this class for introducing base64 encoded constants? For that `Bytes` should be used

Reply:
@jasonpaulos I attempted to address in https://github.com/algorand/pyteal/pull/452.  Can you have a look and resolve if the PR captures it?